src/utils/polygonUtils.cpp

```diff
@@ -329,6 +329,122 @@ unsigned int PolygonUtils::moveInside(const Polygons& polygons, Point& from, int
     return NO_INDEX;
 }
 
+//Version that works on single PolygonRef.
+unsigned int PolygonUtils::moveInside(const ConstPolygonRef polygon, Point& from, int distance, int64_t maxDist2)
```

Comment:
Can this somehow be merged with the `moveInside` above?

Reply:
It's kind of difficult. The reason is documented down below.